src/runtime/graph_executor/graph_executor.cc

```diff
@@ -389,6 +434,13 @@ void GraphExecutor::SetupOpExecs() {
         input_dltensors_[eid].push_back(static_cast<DLTensor*>(op_args->arg_values[i].v_handle));
       }
     }
+    // check if op output is model output
+    if (output_node_id.count(nid) > 0) {
+      for (uint32_t index = inode.inputs.size();
+           index < inode.param.num_outputs + inode.param.num_inputs; ++index) {
+        output_dltensors_.push_back(static_cast<DLTensor*>(op_args->arg_values[index].v_handle));
```

Comment:
for such case  ' "heads": [ [48, 0, 0], [36, 0, 0]], the said logic would wrong.

Reply:
Is it possible to sort outputs_?

Reply:
better to use same way that SetInputZeroCopy used for data tracking and setting.

Reply:
The input index has the same problem, for example "arg_nodes": [3,1,2, 0],

Reply:
do you mind write some comment to address this hidden assumption?

Reply:
I think it is reasonable.
for example arg_nodes:[3,1,2,0]
set_input(0)->node[3]
set_input(1)->node[1]
set_input(2)->node[2]
set_input(3)->node[0]

ref:tvm/tests/python/relay/test_analysis_get_calibration_data.py::test_simple_graph

if sort the input_node_，the ut will fail

Reply:
"sort the output_node" seems like not existing logic, without this new logic,  is the said use case "arg_nodes": [3,1,2, 0]" would cause any UT fail?

Reply:
without sort, ut will not fail.
the UT module heads":[0,0],[1,0],[3,0],[4,0],[3,0],[2,0],[5,0]"
if we want to get node[2] output index is 5, not 3, becauce the outputs_ out of order
node 0 index 0
node 1 index 1
node 3 index 2 or 4
node 4 index 3
node 5 index  6

Reply:
about the disorder solution, SetInputZeroCopy use map of entry_id  and v_handle to fix disorder issue,  how about use similar logic like following here?
//*.h
std::vector<std::vector<DLTensor*>> output_dltensors_;
//*.cc
output_eid = this->entry_id(nid, index - inode.inputs.size())
output_dltensors_[output_eid].push_back(static_cast<DLTensor*>(op_args->arg_values[index].v_handle));

Reply:
It should be no problem, it has been modified